core/src/test/java/com/amplifyframework/core/ShakeDetectorTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+
+import com.amplifyframework.testutils.Resources;
+import com.amplifyframework.testutils.Sleep;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.reactivex.Completable;
+import io.reactivex.schedulers.Schedulers;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests the {@link ShakeDetector} behavior.
+ */
+@RunWith(RobolectricTestRunner.class)
+public class ShakeDetectorTest {
+    // Amount of time in milliseconds between sending sensor
+    // events to the sensor listener.
+    private static final int SENSOR_DATA_DELAY = 65;
```

Comment:
Thanks for writing `milliseconds` in the comment. Even so, I'd bake the unit into the variable name. Ambiguity of time and data storage size units are common, but easily avoidable. e.g. `SENSOR_DATA_POLLING_INTERVAL_MS`?